session_test.go

```diff
@@ -2079,6 +2080,29 @@ func (s *testSessionSuite) TestSessionAuth(c *C) {
 	mustExecSQL(c, se, dropDBSQL)
 }
 
+func (s *testSessionSuite) TestSkipWithGrant(c *C) {
+	defer testleak.AfterTest(c)()
+	dbName := "test_skip_with_grant"
+	se := newSession(c, s.store, dbName)
+	dropDBSQL := fmt.Sprintf("drop database %s;", dbName)
+
+	save1 := privileges.Enable
+	save2 := privileges.SkipWithGrant
```

Comment:
defer can't apply to a expression, it need to use a function.
here is a really simple case, defer seems overkill.